File: euler007-proc.php
#!/usr/bin/env php
<?php
/** Problem 7: 10001st prime
 * https://projecteuler.net/problem=7
 */

function euler007(int $count): int
{
    $primes = [2];
    $candidate = 1;
    while (count($primes) < $count) {
        $candidate += 2; // There's no point in testing even numbers
        foreach($primes as $prime) {
            if ($candidate % $prime == 0) {
                continue 2;
            }
        }
        $primes[] = $candidate;
    }
    return end($primes);
}

// Main
echo euler007(intval(trim(file_get_contents('php://STDIN')))) . PHP_EOL;

File: euler007-proc.py
#!/usr/bin/env python3

# Problem 7: 10001st prime
# https://projecteuler.net/problem=7

import sys

def is_prime(primes, number):
    for prime in primes:
        if number % prime == 0:
            return False
        if prime ** 2 > number:
            break
    return True

def euler007(count):
    primes = [2]
    candidate = 1
    while len(primes) < count:
        candidate += 2
        if is_prime(primes, candidate):
            primes.append(candidate)
    return primes[-1]

def parse_input(lines):
    return int(lines[0].strip())

if __name__ == "__main__":
    print(euler007(parse_input(sys.stdin.readlines())))
